### PR TITLE
Split http and non-http layers

### PR DESCRIPTION
**Pull request by davidfowl**
- This change introduces Microsoft.AspNetCore.SignalR.Http
and Microsoft.AspNetCore.Sockets.Http which expose extension methods
on IAppBuilder for wiring up a sockets and signalr pipeline.

### CODE REVIEW

**Comment by davidfowl** [comment]
@anurse am merging this before @mikaelm12 's ns2.0 changes. Feel free to make changes will I enjoy the 🌞 in hawaii 😄

File: src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionList.cs [hunk 1]
```diff
@@ -34,8 +33,7 @@ public void Add(ConnectionContext connection)
 
         public void Remove(ConnectionContext connection)
         {
-            ConnectionContext dummy;
-            _connections.TryRemove(connection.ConnectionId, out dummy);
+            _connections.TryRemove(connection.ConnectionId, out var dummy);
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionList.cs]: Given that @davidfowl is in Hawaii, I think the only way this nit is being addressed is if you change it yourself :P

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionList.cs]: Hawaii can't be an excuse for flaws of this caliber! ;)

**Comment by davidfowl** [comment]
Oh and can somebody update Coherence.csv

**Pull request merged**